### DISCUSSION TITLE
How to use (get-interpolant)?

### DISCUSSION THREAD

**Discussion by basus** [Z3Prover/z3#5024] [Q&A]
I noticed that the latest version seems to have re-introduced support for getting interpolants. I am confused about how to use the new (get-interpolant) command and how much of Z3's theories it can handle? In particular, I have some code relying on the older (compute-interpolants) with a sequence of arguments. But it seems like (get-interpolant) only supports two arguments? Is it also expected to be much slower than (compute-interpolants)? Some documentation or examples with it would be appreciated.

**Comment by NikolajBjorner** [comment]
This feature isn't really supported until it is used more seriously by @agurfinkel

**Comment by agurfinkel** [comment]
In the mean time, you can use CHC solver to compute interpolants. Pairwise, sequence, and tree interpolants are supported. Should work for arithmetic (linear/real), arrays, bit-vectors (maybe), and ADTs (maybe). Some examples are here: https://github.com/agurfinkel/spacer-on-jupyter/blob/master/Dagstuhl2019.ipynb

**Reply by basus** [reply]
Thank you! I'll take a look at this. Does the reduction to CHC support using Z3's uninterpreted functions and built-in lists?
